cas-server-support-jdbc/src/main/java/org/jasig/cas/adaptors/jdbc/QueryDatabaseAuthenticationHandler.java

```diff
@@ -44,10 +44,10 @@ protected final HandlerResult authenticateUsernamePasswordInternal(final Usernam
         }
 
         final String username = credential.getUsername();
-        final String encryptedPassword = this.getPasswordEncoder().encode(credential.getPassword());
+        final String password = credential.getPassword();
         try {
             final String dbPassword = getJdbcTemplate().queryForObject(this.sql, String.class, username);
-            if (!dbPassword.equals(encryptedPassword)) {
+            if (!getPasswordEncoder().matches(password, dbPassword)) {
```

Comment:
Remembered something. Should other handlers be modified to take advantage of the new method?